app/src/main/java/co/nimblehq/alpine/sample/ui/camera/CameraCaptureActivity.kt

```diff
@@ -126,15 +136,11 @@ class CameraCaptureActivity : ComponentActivity() {
             .setBackpressureStrategy(STRATEGY_KEEP_ONLY_LATEST)
             .build()
         imageAnalysis.setAnalyzer(cameraExecutor) { imageProxy: ImageProxy ->
-            mrzProcessor.processImage(imageProxy.toCameraImage(), object : MrzProcessorResultListener {
-                override fun onSuccess(mrzInfo: MrzInfo) {
-                    imageAnalysis.clearAnalyzer()
-                    imageProxy.close()
-                    NfcScanningActivity.start(this@CameraCaptureActivity, mrzInfo)
-                }
-
-                override fun onError(e: MrzProcessorException) = imageProxy.close()
-            })
+            viewModel.processImage(
+                cameraImage = imageProxy.toCameraImage(),
+                imageProxy = imageProxy,
```

Comment:
We could probably just pass in just the `ImageProxy` here and convert it in the `ViewModel` 😅 
```suggestion
                imageProxy = imageProxy,
```